### PR TITLE
Change to support localhost domain as a valid field for TrustKit-Android

### PR DESCRIPTION
**Pull request by jobot0**
That PR is trying to address the issue #25 
* Making change to support localhost
* Fixing behaviors which could create bugs

### CODE REVIEW

**Comment by jazzyjester** [comment]
Hey @nabla-c0d3 , 
Can you please look on this PR ?
It looks good, didn't checked yet...
Thanks.

File: trustkit/src/main/java/com/datatheorem/android/trustkit/config/DomainPinningPolicy.java [hunk 1]
```diff
@@ -42,7 +43,12 @@
             throws MalformedURLException {
         // Run some sanity checks on the configuration
         // Check if the hostname seems valid
-        DomainValidator domainValidator = DomainValidator.getInstance(false);
+        DomainValidator domainValidator;
+        if (hostname.equals("127.0.0.1") || hostname.equals("localhost")) {
+            domainValidator = DomainValidator.getInstance(true);
+        } else {
+            domainValidator = DomainValidator.getInstance(false);
```
Comment (nabla-c0d3) [inline trustkit/src/main/java/com/datatheorem/android/trustkit/config/DomainPinningPolicy.java]: Why not always use `DomainValidator.getInstance(true)` ? It seems like it would result in the exact same behavior or am I missing something?

File: trustkit/src/main/java/com/datatheorem/android/trustkit/config/TrustKitConfigurationParser.java [hunk 2]
```diff
@@ -160,8 +161,8 @@ private static PinSetTag readPinSet(@NonNull XmlPullParser parser) throws IOExce
     }
 
     private static class TrustkitConfigTag {
-        Boolean enforcePinning = null;
-        Boolean disableDefaultReportUri = null;
+        Boolean enforcePinning = false;
+        Boolean disableDefaultReportUri;
```
Comment (nabla-c0d3) [inline trustkit/src/main/java/com/datatheorem/android/trustkit/config/TrustKitConfigurationParser.java]: Why this change? I think my intent here was to explicitely not initialize these config variables.

File: trustkit/src/main/java/com/datatheorem/android/trustkit/config/TrustKitConfiguration.java [hunk 3]
```diff
@@ -77,7 +77,13 @@ public boolean shouldOverridePins() {
     @Nullable
     public DomainPinningPolicy getPolicyForHostname(@NonNull String serverHostname) {
         // Check if the hostname seems valid
-        DomainValidator domainValidator = DomainValidator.getInstance(false);
+        DomainValidator domainValidator;
+        if (serverHostname.equals("127.0.0.1") || serverHostname.equals("localhost")) {
+            domainValidator = DomainValidator.getInstance(true);
+        } else {
+            domainValidator = DomainValidator.getInstance(false);
+        }
```
Comment (nabla-c0d3) [inline trustkit/src/main/java/com/datatheorem/android/trustkit/config/TrustKitConfiguration.java]: Same question as above.

File: trustkit/src/main/java/com/datatheorem/android/trustkit/config/DomainPinningPolicy.java (hunk 1 above)
Comment (jobot0) [reply trustkit/src/main/java/com/datatheorem/android/trustkit/config/DomainPinningPolicy.java]: The main is difference is :
https://github.com/datatheorem/TrustKit-Android/blob/3f96ec34d5a1fd8a3bcf24bb7c572dc4fe48170a/trustkit/src/main/java/com/datatheorem/android/trustkit/config/DomainValidator.java#L121-L126

File: trustkit/src/main/java/com/datatheorem/android/trustkit/config/TrustKitConfiguration.java (hunk 3 above)
Comment (jazzyjester) [reply trustkit/src/main/java/com/datatheorem/android/trustkit/config/TrustKitConfiguration.java]: @nabla-c0d3 
https://github.com/datatheorem/TrustKit-Android/blob/3f96ec34d5a1fd8a3bcf24bb7c572dc4fe48170a/trustkit/src/main/java/com/datatheorem/android/trustkit/config/DomainValidator.java#L155

Please note that the `allowLocal` flag should be true and it's false without this change...

File: trustkit/src/main/java/com/datatheorem/android/trustkit/config/DomainPinningPolicy.java (hunk 1 above)
Comment (jazzyjester) [reply trustkit/src/main/java/com/datatheorem/android/trustkit/config/DomainPinningPolicy.java]: @jobot0 probably can be inited in one line
domainValidator = DomainValidator.getInstance(hostname.equals("127.0.0.1") || hostname.equals("localhost"));
but it's matter of style...
